### PR TITLE
[Merged by Bors] - feat: Fin CharP and lemmas for Fin rollover

### PR DESCRIPTION
**Pull request by pechersky**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Data/Fin/Basic.lean [hunk 1]
```diff
@@ -1324,6 +1324,54 @@ theorem last_sub (i : Fin (n + 1)) : last n - i = Fin.rev i :=
...
+theorem neg_nat_cast_eq_one (n : ℕ) : -(n : Fin (n + 1)) = 1 := by
+  simp [neg_eq_iff_add_eq_zero, ext_iff, val_add, cast_nat_eq_last]
+
+lemma pos_of_ne_zero {n : ℕ} {a : Fin (n + 1)} (h : a ≠ 0) :
+    0 < a :=
+  Nat.pos_of_ne_zero (val_ne_of_ne h)
+
+@[simp]
+theorem nat_cast_add_one_eq_zero (n : ℕ) : (n : Fin (n + 1)) + 1 = 0 := by
+  rw [← neg_nat_cast_eq_one, ← sub_eq_add_neg, sub_self]
```
Comment (alreadydone) [inline Mathlib/Data/Fin/Basic.lean:1373]: Linter not happy with these now
```
/- The `simpNF` linter reports:
SOME SIMP LEMMAS ARE NOT IN SIMP-NORMAL FORM.
see note [simp-normal form] for tips how to debug this.
https://leanprover-community.github.io/mathlib_docs/notes.html#simp-normal%[20](https://github.com/leanprover-community/mathlib4/actions/runs/7335822631/job/19974251086#step:20:21)form -/
-- Mathlib.Data.Fin.Basic
Error: ./././Mathlib/Data/Fin/Basic.lean:1362:1: error: Fin.neg_nat_cast_eq_one simp can prove this:
  by simp only [Fin.cast_nat_eq_last, Fin.neg_last]
One of the lemmas above could be a duplicate.
If that's not the case try reordering lemmas or adding @[priority].

Error: ./././Mathlib/Data/Fin/Basic.lean:1370:1: error: Fin.nat_cast_add_one_eq_zero simp can prove this:
  by simp only [Fin.cast_nat_eq_last, Fin.last_add_one]
One of the lemmas above could be a duplicate.
If that's not the case try reordering lemmas or adding @[priority].
```

Comment (pechersky) [reply Mathlib/Data/Fin/Basic.lean:1371]: Yury modified `cast_nat_eq_last` and made it a simp lemma.

**Comment by pechersky** [comment]
bors merge

**Comment by alreadydone** [comment]
Let's get it merge by retriggering CI with added label.

**Pull request closed without merging**